_inc/lib/core-api/wpcom-endpoints/service-api-keys.php

```diff
@@ -268,6 +284,61 @@ public static function validate_service_api_key_mapbox( $key ) {
 		);
 	}
 
+	/**
+	 * Get the site's own Mapbox API key if set, or the Automattic's one otherwise.
+	 *
+	 * @return array An array containing the key (if any) and its source ("site" or "automattic").
+	 */
+	public static function get_service_api_key_mapbox() {
+		$option          = self::key_for_api_service( 'mapbox' );
+		$service_api_key = Jetpack_Options::get_option( $option, '' );
+
+		// If the site provides its own Mapbox API key, return it.
+		if ( ! empty( $service_api_key ) ) {
+			return array(
+				'key'    => $service_api_key,
+				'source' => 'site',
+			);
+		}
+
+		// If the site is not WPCOM, return an empty API key.
+		if ( ! ( defined( 'IS_WPCOM' ) && IS_WPCOM ) && ! jetpack_is_atomic_site() ) {
+			return array(
+				'key'    => '',
+				'source' => 'site',
+			);
+		}
+
+		$transient_key = 'mapbox_a8c_access_token';
+		$cached_token  = get_transient( $transient_key );
+
+		// If there is a cached token, return it.
+		if ( ! empty( $cached_token ) ) {
+			return $cached_token;
+		}
+
+		// Otherwise retrieve an Automattic token.
+		$site_id  = Jetpack_Options::get_option( 'id' );
+		$response = Client::wpcom_json_api_request_as_blog( sprintf( '/sites/%d/mapbox', $site_id ), '2', array(), null, 'wpcom' );
+
+		if ( 200 !== wp_remote_retrieve_response_code( $response ) ) {
+			return array(
+				'key'    => '',
+				'source' => 'site',
+			);
+		}
+		$response_body           = json_decode( wp_remote_retrieve_body( $response ) );
+		$mapbox_a8c_access_token = $response_body->mapbox_a8c_access_token;
+
+		// Cache the Automattic token for a month.
+		set_transient( $transient_key, $mapbox_a8c_access_token, 2592000 );
```

Comment:
I think a month may be too much if we have to rotate the key. How about something like an hour, or at max a day?